target/ppc/translate/storage-ctrl-impl.c.inc

```diff
@@ -0,0 +1,93 @@
+/*
+ * Power ISA decode for Storage Control instructions
+ *
+ * Copyright (c) 2022 Instituto de Pesquisas Eldorado (eldorado.org.br)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * Store Control Instructions
+ */
+
+static bool do_tlbie(DisasContext *ctx, arg_X_tlbie *a, bool local)
+{
+#if defined(CONFIG_USER_ONLY)
+    GEN_PRIV2(true);
+#else
+    TCGv_i32 t1;
+    int rb, ric;
+    bool prs, r;
+
+    rb = a->rb;
+    ric = a->ric;
+    prs = a->prs;
+    r = a->r;
+
+    if (ric || prs || r) {
```

Comment:
```suggestion
    if (ric || prs || r || (local && a->rs)) {
```

Just noticed that tlbiel before 3.00 didn't support RS too. So this need to be changed too.

Reply:
This case is now being handled in the comment below, by setting a->rs to 0 if local, when ISA < 3.00.